agh_graphs/productions/p8.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+
+from networkx import Graph
+from agh_graphs.production import Production
+from utils import get_neighbors_at, find_overlapping_vertices, join_overlapping_vertices
+
+
+class P8(Production):
+
+    def apply(self, graph: Graph, prod_input: List[str], orientation: int = 0, **kwargs) -> List[str]:
+
+        self.check_input_graph(graph, prod_input)
+
+        overlapping_vertices = find_overlapping_vertices(graph)
+
+        join_overlapping_vertices(graph, overlapping_vertices[0][0],
```

Comment:
Wydaje mi się że to nie do końca będzie działać :thinking: nie używamy tutaj w ogóle nazw wierzchołków z `prod_input` a na ich podstawie powinnaś wybrać te dwa wierzchołki E które trzeba połączyć, czyż nie? Bierzesz tutaj pierwszą lepszą parę nachodzących na siebie wierzchołków z grafu.